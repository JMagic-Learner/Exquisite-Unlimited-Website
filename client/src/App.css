.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.navigation-flex-row{
  margin-left:50%;
}

.nav-item{
  float:right;
}

.Banner-Container{
  width: 100%;
}

.Banner-Image-Container {
  height:50vw;
  background-position: center;
}

.banner-overlay {
  z-index:1;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.tab-text{
  z-index:2;
  position: relative;
  top:50%;
  left: 50%;
  color:white;
  transform: translate(-50%, 30%);
  transition: color 1s ease-in-out
}


.tabs-inner {
  width:100%;
}
/* .tab-background {
  margin:1%;
} */

.Tab-Image-Container{
  margin:1%;
  height:300px;
  opacity:0.1;
  z-index:-1;
  transition: opacity 0.5s ease-in-out;
  background-position: center;
}

.Tab-Image-Container:hover{
  margin:1%;
  height:300px;
  opacity:1;
  z-index:-1;
  transition: opacity 0.5s ease-in-out;
  background-position: center;
}


.horizontal-container{
  width:95%;
  margin:auto;
}

.homepage-container-lower {
margin-top:1%;
}


@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* ABOUT */

.about{
  margin-top:5%;
}

.paragraph{
  width:60%;
  margin-left:10vw;
}
.column-image{
  height:1000px;
  background-position: center;
}

/* HOVER ITEM */
.hover-action{
  background-color: white;
  width:100px;
  transition: background-color ease-out 0.5s;
}

.hover-action:hover{
  background-color: #e4e4e4;
  width:100px;
  transition: background-color ease-in 0.5s;
}

/* Shop Menu */
.shop-menu {
  width:75%;
  margin:auto;
}

.shop-menu-button {
  width:300px;
}

@media only screen and (max-width: 970px) {
  .tab-text{
    z-index:2;
    position: relative;
    /* top:50%;
    left: 50%; */
    color:black;
    transform: translate(-50%, 50%);
  }
  .banner-overlay {
    z-index:1;
    position: absolute;
    top: 15%;
    left: 50%;
    transform: translate(-50%, -50%);
  }
  .Banner-Image-Container {
    right:50%;
  }
  .column-image{
    margin-top:25%;
    height:1300px;
    background-position: center;
  }
}
/* @media only screen and (max-width: 1380px) {
  .Banner-Image-Container {
    height:1000px;
    width:100%;
    position:relative;
    margin:auto;
  }
} */

.catalogue{
  width:80%;
  display:flex;
  margin:auto;
}

.product-card{
  width:50%;
  margin:1%;
  background-color:ivory;
  box-shadow:inset;
}

.product-description{
  font-size: 1.5vh;
}
