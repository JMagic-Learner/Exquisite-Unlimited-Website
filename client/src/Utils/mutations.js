import { gql } from '@apollo/client';

export const BUY_PRODUCT = gql`
  mutation buyProduct($_id: ID!, $quantity: Int!) {
    buyProduct(product: id:$_id, quantity: $quantity) {
      
      product {
        _id
        quantity
      }
    }
  }
`;