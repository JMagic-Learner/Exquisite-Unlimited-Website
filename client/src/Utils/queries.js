import { gql } from '@apollo/client';


export const QUERY_PRODUCTS = gql`
  query getProducts {
    products {
      _id
      name
      category
      description
      price
      serial
      quantity
      size
    }
  }`

  export const QUERY_SINGLE_PRODUCT = gql`
  query getProduct($name: String!) {
    product(name: $name){
        _id
        name
        category
        description
        price
        serial
        quantity
        size
    }
  }`