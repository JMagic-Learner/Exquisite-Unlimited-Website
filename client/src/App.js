import logo from './logo.svg';
import './App.css';
import Navigation from './Components/Navigation';
import Home from './Pages/HomePage'
import About from './Pages/AboutPage'
import Shop from './Pages/ShopPage';
import {  BrowserRouter, Routes, Route} from 'react-router-dom';
import {
  ApolloClient,
  InMemoryCache,
  ApolloProvider,
  createHttpLink,
  gql
} from '@apollo/client';
import { setContext } from '@apollo/client/link/context';

const authLink = setContext((_, { headers }) => {
  // if (NODE_ENV=="Production") {
  //   return "https://exquisite-unlimited-website.herokuapp.com/graphql"
  // } else {
  //   return "http://localhost:4000/shop"
  // }

});


const httpLink = createHttpLink({
  uri: 'https://exquisite-unlimited-website.herokuapp.com/graphql',
  // uri: `http://localhost:27017/graphql`,
});

const client = new ApolloClient({
  // Set up our client to execute the `authLink` middleware prior to making the request to our GraphQL API
  link: httpLink,
  cache: new InMemoryCache(),
  playground: true,
  introspection: true,
});

function App() {
  return (
    <div className="App">
        <ApolloProvider client={client}>
      <BrowserRouter>
      <Navigation/>
      <Routes>
            <Route exact path="/" element={<Home />}/>
            <Route exact path="/about" element={<About />}/>
            <Route exact path="/shop" element={<Shop/>}/>
       </Routes>
       </BrowserRouter>
       </ApolloProvider>

    </div>
  );
}

export default App;
