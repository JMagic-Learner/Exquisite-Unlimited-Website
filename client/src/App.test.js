import { render, screen } from '@testing-library/react';
import { MockedProvider } from "@apollo/client/testing";
import {BrowserRouter, MemoryRouter} from 'react-router-dom'
import {QUERY_PRODUCTS} from './Utils/queries'
import App from './App';
import Shop from './Pages/ShopPage';
import About from './Pages/AboutPage';


const mocks = [
  {
    request: {
      query: QUERY_PRODUCTS,
    result: {
      products: {
        products: { pictureID: 1, name: "Test", category: "Oil", description: "Oil", price: 50.00, serial: "xx1", quantity:1, size:"test"}
      }
    }
  }
}
];

test('Renders default application UI', () => {
  render(
    <MockedProvider mocks={mocks} addTypename={false}> 
    <App />
      </MockedProvider>
 );
  const linkElement = screen.getByText("Exquisite Unlimited");
  const NAVBAR = screen.getByText('Home')
  expect(linkElement).toBeInTheDocument();
  expect(NAVBAR).toBeInTheDocument();
});

test('Landing on About Page', () => {
  const AboutRoute = '/about'
  render(
    <MockedProvider mocks={mocks} addTypename={false}> 
    <App>
      <MemoryRouter initialEntries={[AboutRoute]}>
        <About />
      </MemoryRouter>,
    </App>
      </MockedProvider>
  )
});


describe ( `A Snapshot of the About Page`, ()=> {
  it('should match snapshot', () => {
    })
    const { asFragment } = render(<About > SnapShot Rewardpage </About>);
    expect(asFragment()).toMatchSnapshot();
})

test('Describe the About Page as it should appear', () => {

  render(
        <About />
  )

  const Title = screen.getByText(`"Life revolves around Color"`)
  expect(Title).toBeInTheDocument();
  const SubTitle= screen.getByText(`2008, The world turned upside down`)
  expect(SubTitle).toBeInTheDocument();
  const SubTitle2= screen.getByText(`Our Products`)
  expect(SubTitle2).toBeInTheDocument();
});
