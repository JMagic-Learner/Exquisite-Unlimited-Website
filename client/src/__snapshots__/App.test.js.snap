// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="About-Container"
  >
    <h1
      class="text-black font-italic"
    >
       "Life revolves around Color" 
    </h1>
    <div
      class="row"
    >
      <div
        class="col column-image"
        style="background-image: url(about-column.jpg);"
      />
      <div
        class="col"
      >
        <section
          class="about"
        >
          <h2>
             2008, The world turned upside down 
          </h2>
          <div
            class="paragraph text-justify"
          >
            Faced with a world-economy that was free-falling, we saw opportunity to plant seeds for a long term goal. We wanted to bring color to households, on an affordable price tag.
          </div>
        </section>
        <section
          class="about"
        >
          <h2>
             Our Products 
          </h2>
          <div
            class="paragraph text-justify"
          >
            Exquisite Unlimited LLC carries three primary types of paintings
          </div>
        </section>
        <section
          class="about"
        >
          <h4>
             Engravings 
          </h4>
          <div
            class="paragraph text-justify"
          >
            Engravings are paintings that incorporate 3D-topography. Traditional paintings are purely 2d, but our Engraving product line are sculpted murals to be hanged on your walls. Made with a variety of metals, these paintings weigh the heaviest. Subjects of this product line range from religious iconography, to mythical landscapes and creatures.
          </div>
        </section>
        <section
          class="about"
        >
          <h4>
             Oils
          </h4>
          <div
            class="paragraph text-justify"
          >
            Our Oil paintings sport the most vibrant color assortment of our product range. Depicting villas, cities, fields, forests and more in vivid colors, Oils are our main product line.
          </div>
        </section>
        <section
          class="about"
        >
          <h4>
             Contemporary 
          </h4>
          <div
            class="paragraph text-justify"
          >
            The Contemporary product line highlights art styles from the 1980's onward. This product range includes abstract expressionism,modernism, surrealism, geometric etc
          </div>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
